lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -1671,15 +1671,24 @@ SECOND_PASS:
     void JavascriptArray::EnsureCalculationOfAllocationBuckets()
     {
         // If allocation size for current ArrayType has already calculated, no need to recalculate it.
-        bool determineAllocationSize = ArrayType::allocationBuckets[0][MissingElementsCountIndex] == 0;
-        if (determineAllocationSize)
+#if defined(_M_ARM32_OR_ARM64)
+        // On ARM machines, order of writes is not guaranteed. So we need to have one memory barrier
+        // before reading the IsArrayAllocationBucketsInitialized and another one after writing
+        // to the IsArrayAllocationBucketsInitialized.
+        MemoryBarrier();
+#endif
+        if (!ArrayType::IsArrayAllocationBucketsInitialized)
```

Comment:
Yeah, i think thats much better and cleaner. I was deferring the initialization till we actually have to use those values. But doesn't hurt if we initialize one-time in `JavascriptLibrary` for all array types. I will update PR. Thanks for the feedback.